src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnectionFactory.cs

```diff
@@ -86,23 +87,22 @@ internal static HttpConnectionOptions ShallowCopyHttpConnectionOptions(HttpConne
             {
                 HttpMessageHandlerFactory = options.HttpMessageHandlerFactory,
                 Headers = options.Headers,
-                Cookies = options.Cookies,
                 Url = options.Url,
                 Transports = options.Transports,
                 SkipNegotiation = options.SkipNegotiation,
                 AccessTokenProvider = options.AccessTokenProvider,
                 CloseTimeout = options.CloseTimeout,
-                Credentials = options.Credentials,
-                Proxy = options.Proxy,
-                UseDefaultCredentials = options.UseDefaultCredentials,
                 DefaultTransferFormat = options.DefaultTransferFormat,
                 WebSocketConfiguration = options.WebSocketConfiguration,
             };
 
-            // WASM doesn't support Crypto APIs and our setter throws if you try to assign null
-            if (options.ClientCertificates != null)
+            if (!OperatingSystem.IsBrowser())
             {
+                newOptions.Cookies = options.Cookies;
                 newOptions.ClientCertificates = options.ClientCertificates;
+                newOptions.Credentials = options.Credentials;
+                newOptions.Proxy = options.Proxy;
+                newOptions.UseDefaultCredentials = options.UseDefaultCredentials;
```

Comment:
Why go out of are way not to include these properties in the shallow copy?

Reply:
I don't think setting these properties throw. This our type. The HttpConnectionOptions.UseDefaultCredentials setter shouldn't throw on any platform. I don't think the other property setters would throw either unless it throws an ArgumentNullException which has nothing to do with running in the browser.

https://github.com/dotnet/aspnetcore/blob/1c200f24d339014045bcf6a635df7897106f3447/src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnectionOptions.cs#L118

Reply:
Maybe you thought this was the HttpClientHandler? This PR already guards against setting unsettable properties on that in the browser.

Reply:
Ok. Now I see the HttpConnectionOptions change. 😆 Resolving.